test/doc/traits/callable_object.cpp

```diff
@@ -0,0 +1,40 @@
+#include <iostream>
+#include <type_traits>
+#include <eve/traits/overload.hpp>
+#include <eve/wide.hpp>
+
+namespace eve
+{
+  struct func_t : basic_callable<func_t>
+  {
+    // Supported signatures
+    template<eve::integral_value T> auto call(T const&) -> T;
+                                    auto call(double)   -> double;
+                                    auto call(float)    =  delete;
+
+    // Register this callable type to EVE overload system
+    EVE_CALLABLE_OBJECT(func_t, func_);
+  } inline constexpr func;
+};
+
+// As func_t used EVE_CALLABLE_OBJECT, we should write overloads in eve::detail
+namespace eve::detail
+{
+  auto func_(EVE_REQUIRES(cpu_), auto, eve::integral_value auto x) {  return x*x;  }
```

Comment:
auto is options? Can you specify that?

Reply:
it'll be somethign like `rbr::concepts::settings auto`